### PR TITLE
Alpine docker

### PR DESCRIPTION
**Pull request by sysadmiral**
I've based the image on alpine to reduce the size of the image from ~800MB to ~300MB.

I've done some testing with the following repos:
- ruby - jekyll
- python - python-mysqltuner
- javascript - You-Dont-Need-Lodash-Underscore
- php - composer

Let me know if there are any issues with the Dockerfile/image.

Cheers,

Amo

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,30 @@
-FROM jruby:9.0.3-jdk
+FROM alpine:latest
```
Comment (pbrisbin) [inline Dockerfile]: Using a `latest` tag runs the risk of non-reproducible builds as time goes on. Can you use the latest released tag, I think it's `3.4`?

**Comment by pbrisbin** [comment]
Ping @mrb do you remember the historical context around the JRuby choice here? It's unclear if we had a concrete performance issue that was addressed by moving to JRuby.

**Comment by pbrisbin** [comment]
I could the commit where we switched to JRuby: https://github.com/codeclimate/codeclimate-duplication/commit/4d341576a3fedc8e25a194edfb10a7d8fb83b7f5#diff-3254677a7917c6c01f55212f86c57fbf it includes some benchmarks for "a medium sized JavaScript project".

File: Dockerfile (hunk 1 above)
Comment (sysadmiral) [inline Dockerfile]: updated! :)

**Comment by sysadmiral** [comment]
I see. I did wonder about the decision since most other engines Dockerfiles used alpine.

If it would be better to continue with a larger but faster engine then this PR can be closed?

**Comment by pbrisbin** [comment]
> If it would be better to continue with a larger but faster engine then this PR can be closed?

I'm not sure. We definitely appreciate your effort here, so we wanted to look into the performance implications before just closing the PR. That said, there may be a bit _more_ effort involved in pinning down those performance questions, and living the larger image is not a problem with us.

So, if you're willing to work on getting some performance metrics on various projects (especially larger ones) we'd be happy to evaluate that and (if the performance loss isn't great) work towards merging this. If you'd rather not though, we'll stick with the larger JRuby image for now.

**Comment by sysadmiral** [comment]
I'd be happy to compare analysis results on a set of repos (top 50 trending ruby repos on github?) but do I get the timings just using the `--dev` flag?

**Comment by pbrisbin** [comment]
I think I would use this approach:

- Build an image from master with a tag (e.g. `:jruby`)
- Build an image from this branch with another tag (e.g. `:alpine`)
- In each example project:
- Tag the jruby image as `codeclimate/codeclimate-duplication`
- Run `time codeclimate analyze -e duplication`
- Tag the alpine image as `codeclimate/codeclimate-duplication`
- Run `time codeclimate analyze -e duplication`

**Comment by sysadmiral** [comment]
Great I will come back here when I have more information. Thanks for the help! :smile:

**Comment by dblandin** [comment]
Hey @sysadmiral,

Just checking in. :) Any update on this PR?

**Comment by sysadmiral** [comment]
Sorry not yet. I should have time next week though to get some of this done. Would that be ok?

**Comment by dblandin** [comment]
> Sorry not yet. I should have time next week though to get some of this done. Would that be ok?

For sure! Ping us when you're ready for another review. Thanks!

**Comment by dblandin** [comment]
Hey @sysadmiral. Any interest in finishing this up? If not, no worries. We can revisit this at a later time.

**Comment by sysadmiral** [comment]
Hi @dblandin definitely got the interest just struggling to get the time right now.

I'm happy for you to close this if it helps your "housekeeping" but if you are ok to leave it open for me to work on it that would be great.

Let me know what works best for you.

**Comment by dblandin** [comment]
Going to go ahead and close this PR for now. Feel free to reopen later!

**Pull request closed without merging**